core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/OnDemandRepairSchedulerImpl.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2020 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.core.repair;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.bss.cassandra.ecchronos.core.JmxProxyFactory;
+import com.ericsson.bss.cassandra.ecchronos.core.metrics.TableRepairMetrics;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.state.ReplicationState;
+import com.ericsson.bss.cassandra.ecchronos.core.scheduling.ScheduleManager;
+import com.ericsson.bss.cassandra.ecchronos.core.scheduling.ScheduledJob;
+import com.ericsson.bss.cassandra.ecchronos.core.utils.TableReference;
+
+/**
+ * A factory creating {@link OnDemandRepairJob}'s for tables.
+ */
+public class OnDemandRepairSchedulerImpl implements OnDemandRepairScheduler, Closeable
+{
+    private static final Logger LOG = LoggerFactory.getLogger(OnDemandRepairSchedulerImpl.class);
+
+    private final Map<TableReference, OnDemandRepairJob> myScheduledJobs = new HashMap<>();
```

Comment:
I think we should use a UUID instead of a TableReference.

As soon as we get a scheduling feature on a cluster level it will become important to keep track of this.

Additionally this should help with the integration test.

Reply:
Done